types/model.d.ts

```diff
@@ -74,9 +115,135 @@ export interface ModelProperties {
  * }
  * ```
  */
-export interface ModelSettings extends AnyObject {
+export interface ModelSettings extends AnyObject, ModelUtilsOptions {
   strict?: boolean;
+
+  /**
+   * Set if manual assignment of auto-generated ID values should be blocked.
+   */
   forceId?: boolean;
+
+  /**
+   * @remarks
+   * Defaults to `true`.
+   */
+  idInjection?: boolean;
+
+  plural?: string;
+
+  http?: {
+    path?: string;
+  }
+
+  /**
+   * @remarks
+   * Alias of {@link ModelSettings.super}. Takes lower precedence.
+   */
+  base?: ModelBaseClass;
+  /**
+   * @remarks
+   * Alias of {@link ModelSettings.base}. Takes lower precedence.
+   */
+  super?: ModelBaseClass;
+  excludeBaseProperties?: string[];
+
+  /**
+   * Indicates if the {@link ModelBaseClass | Model} is attached to the
+   * DataSource
+   * @internal
+   */
+  unresolved?: boolean;
+
+  indexes?: {
+    [indexJugglerName: string]: IndexDefinition
+  };
+
+  foreignKeys?: {
+    [fkJugglerName: string]: {
+      name: string,
+      entity: ModelBase | string,
+      entityKey: string,
+      foreignKey: string,
+      onDelete?: string,
+      onUpdate?: string,
+    }
+  }
+
+  /**
+   * {@inheritDoc ModelSettings.tableName}
+   */
+  tableName?: string
+
+  /**
+   * Mapped table name for the model.
+   */
+  table?: string;
+
+  /**
+   * Sets if JavaScript {@link undefined} as an attribute value should be
+   * persisted as database `NULL`.
+   */
+  persistUndefinedAsNull?: boolean;
+
+  /**
+   * Model properties to be set as protected.
+   *
+   * @remarks
+   * Protected properties are not serialised to JSON or {@link object} when the
+   * model is a nested child.
+   *
+   * Mostly used by {@link ModelBase.toObject}.
+   */
+  protectedProperties?: string[];
+
+  /**
+   * {@inheritDoc ModelSettings.protectedProperties}
+   *
+   * @remarks
+   * Overriden by {@link ModelSettings.protectedProperties}.
+   *
+   * @deprecated Use {@link ModelSettings.protectedProperties} instead.
+   */
+  protected?: string[];
+
+  /**
+   * Model properties to be set as hidden.
+   *
+   * @remarks
+   * Hidden properties are
+   */
+  hiddenProperties?: string[];
+
+  /**
+   * {@inheritDoc ModelSettings.hiddenProperties}
+   *
+   * @remarks
+   * Overriden by {@link ModelSettings.hiddenProperties}.
+   *
+   * @deprecated Use {@link ModelSettings.hiddenProperties} instead.
+   */
+  hidden?: string[];
+  automaticValidation?: boolean;
+  updateOnLoad?: boolean;
+  validateUpsert?: boolean;
+
+  /**
+   * Sets if an {@link Error} should be thrown when no instance(s) were found
+   * for delete operation.
+   *
+   * @remarks
+   *
+   * This setting is used by these operations:
+   *
+   * - {@link DataAccessObject.removeById}/{@link DataAccessObject.destroyById}/{@link DataAccessObject.deleteById}
+   * - {@link DataAccessObject.remove}/{@link DataAccessObject.delete}/{@link DataAccessObject.destroy}
+   */
+  strictDelete?: boolean;
```

Comment:
What's the default value? IIRC, it's `false` as LoopBack 4's `DefaultCrudRepository` explicitly sets this to `true`.